### ISSUE TITLE
System.NullReferenceException: Object reference not set to an instance of an object.

### ISSUE THREAD

**Issue by ademclk** [dotnet/efcore#30295]
## Mapping aggregates to database

I was mapping my aggregate roots to my database. Made 9 of them without errors but the last one I couldn't make it. Everything seems same but I couldn't figure out.

If you want to check my project files to understand what's going on [ademclk/ceyehatapi](https://github.com/ademclk/ceyehat-api)

If you check my city configurations file I mapped entities like this and it worked fine. I don't know what's the problem with this one.

Help needed. If you want to check issue [ademclk/ceyehatapi/issues](https://github.com/ademclk/ceyehat-api/issues/60)

Passenger configurations file.

```C#
using Ceyehat.Domain.CustomerAggregate.ValueObjects;
using Ceyehat.Domain.PassengerAggregate;
using Ceyehat.Domain.PassengerAggregate.Entities;
using Ceyehat.Domain.PassengerAggregate.ValueObjects;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace Ceyehat.Infrastructure.Persistence.Configurations;

public class PassengerConfigurations : IEntityTypeConfiguration<Passenger>
{
    public void Configure(EntityTypeBuilder<Passenger> builder)
    {
        ConfigurePassengersTable(builder);
        ConfigurePassengerFlightTicketsTable(builder);
        ConfigurePassengerBoardingPassesTable(builder);
    }

    private void ConfigurePassengerBoardingPassesTable(EntityTypeBuilder<Passenger> builder)
    {
        builder.OwnsMany(
            p => p.BoardingPasses,
            tb =>
            {
                tb.ToTable("PassengerBoardingPasses");

                tb.HasKey(b => b.Id);

                tb.Property(b => b.Id)
                    .HasColumnName("PassengerBoardingPassId")
                    .ValueGeneratedNever()
                    .HasConversion(
                        id => id.Value,
                        value => BoardingPassId.Create(value));

                tb.Property(b => b.FlightTicketId)
                    .ValueGeneratedNever()
                    .HasConversion(
                        id => id.Value,
                        value => FlightTicketId.Create(value));

                tb.Property(b => b.Gate)
                    .HasMaxLength(8);

                tb.Property(b => b.BoardingTime)
                    .IsRequired();
            });
    }

    private void ConfigurePassengerFlightTicketsTable(EntityTypeBuilder<Passenger> builder)
    {
        builder.OwnsMany(
            p => p.FlightTickets,
            tb =>
            {
                tb.ToTable("PassengerFlightTickets");

                tb.Property(t => t.Id)
                    .HasColumnName("PassengerFlightTicketId")
                    .ValueGeneratedNever()
                    .HasConversion(
                        id => id.Value,
                        value => FlightTicketId.Create(value));

                tb.Property(t => t.BookingId)
                    .ValueGeneratedNever()
                    .HasConversion(
                        id => id.Value,
                        value => BookingId.Create(value));
            });
    }

    private void ConfigurePassengersTable(EntityTypeBuilder<Passenger> builder)
    {
        builder.ToTable("Passengers");

        builder.HasKey(p => p.Id);

        builder.Property(p => p.Id)
            .ValueGeneratedNever()
            .HasConversion(
                p => p.Value,
                value => PassengerId.Create(value));

        builder.Property(p => p.CustomerId)
            .ValueGeneratedNever()
            .HasConversion(
                p => p.Value,
                value => CustomerId.Create(value));

    }
}
```

### Stack Trace

```
Build started...
Build succeeded.
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionBatchExtensions.Run(IConventionBatch batch, InternalForeignKeyBuilder relationshipBuilder)
   at Microsoft.EntityFrameworkCore.Metadata.Internal.InternalForeignKeyBuilder.ReuniquifyImplicitProperties(Boolean force)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.ForeignKeyPropertyDiscoveryConvention.DiscoverProperties(IConventionForeignKeyBuilder relationshipBuilder, IConventionContext context)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.ForeignKeyPropertyDiscoveryConvention.ProcessForeignKeyRequirednessChanged(IConventionForeignKeyBuilder relationshipBuilder, IConventionContext`1 context)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.ImmediateConventionScope.OnForeignKeyRequirednessChanged(IConventionForeignKeyBuilder relationshipBuilder)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.OnForeignKeyRequirednessChangedNode.Run(ConventionDispatcher dispatcher)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.DelayedConventionScope.Run(ConventionDispatcher dispatcher)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.ConventionBatch.Run()
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.ConventionBatch.Run(IConventionForeignKey foreignKey)
   at Microsoft.EntityFrameworkCore.Metadata.Builders.EntityTypeBuilder`1.OwnsManyBuilder[TRelatedEntity](TypeIdentity ownedType, MemberIdentity navigation)
   at Microsoft.EntityFrameworkCore.Metadata.Builders.EntityTypeBuilder`1.OwnsMany[TRelatedEntity](Expression`1 navigationExpression, Action`1 buildAction)
   at Ceyehat.Infrastructure.Persistence.Configurations.PassengerConfigurations.ConfigurePassengerFlightTicketsTable(EntityTypeBuilder`1 builder) in /Users/ademclk/RiderProjects/ceyehat-api/Ceyehat.Infrastructure/Persistence/Configurations/PassengerConfigurations.cs:line 51
   at Ceyehat.Infrastructure.Persistence.Configurations.PassengerConfigurations.Configure(EntityTypeBuilder`1 builder) in /Users/ademclk/RiderProjects/ceyehat-api/Ceyehat.Infrastructure/Persistence/Configurations/PassengerConfigurations.cs:line 15
   at Microsoft.EntityFrameworkCore.ModelBuilder.ApplyConfiguration[TEntity](IEntityTypeConfiguration`1 configuration)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.EntityFrameworkCore.ModelBuilder.ApplyConfigurationsFromAssembly(Assembly assembly, Func`2 predicate)
   at Ceyehat.Infrastructure.Persistence.CeyehatDbContext.OnModelCreating(ModelBuilder modelBuilder) in /Users/ademclk/RiderProjects/ceyehat-api/Ceyehat.Infrastructure/Persistence/CeyehatDbContext.cs:line 37
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelCustomizer.Customize(ModelBuilder modelBuilder, DbContext context)
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.CreateModel(DbContext context, IConventionSetBuilder conventionSetBuilder, ModelDependencies modelDependencies)
   at Microsoft.EntityFrameworkCore.Infrastructure.ModelSource.GetModel(DbContext context, ModelCreationDependencies modelCreationDependencies, Boolean designTime)
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.CreateModel(Boolean designTime)
   at Microsoft.EntityFrameworkCore.Internal.DbContextServices.get_Model()
   at Microsoft.EntityFrameworkCore.Infrastructure.EntityFrameworkServicesBuilder.<>c.<TryAddCoreServices>b__8_4(IServiceProvider p)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.<>c__DisplayClass2_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()
   at Microsoft.EntityFrameworkCore.DbContext.get_ContextServices()
   at Microsoft.EntityFrameworkCore.DbContext.get_InternalServiceProvider()
   at Microsoft.EntityFrameworkCore.DbContext.Microsoft.EntityFrameworkCore.Infrastructure.IInfrastructure<System.IServiceProvider>.get_Instance()
   at Microsoft.EntityFrameworkCore.Infrastructure.Internal.InfrastructureExtensions.GetService[TService](IInfrastructure`1 accessor)
   at Microsoft.EntityFrameworkCore.Infrastructure.AccessorExtensions.GetService[TService](IInfrastructure`1 accessor)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(Func`1 factory)
   at Microsoft.EntityFrameworkCore.Design.Internal.DbContextOperations.CreateContext(String contextType)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.AddMigration(String name, String outputDir, String contextType, String namespace)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigrationImpl(String name, String outputDir, String contextType, String namespace)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigration.<>c__DisplayClass0_0.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
Exception has been thrown by the target of an invocation.

```

### Verbose output

```
Using project '/Users/ademclk/RiderProjects/ceyehat-api/Ceyehat.Infrastructure/Ceyehat.Infrastructure.csproj'.
Using startup project '/Users/ademclk/RiderProjects/ceyehat-api/ceyehat/ceyehat.csproj'.
Writing '/Users/ademclk/RiderProjects/ceyehat-api/Ceyehat.Infrastructure/obj/Ceyehat.Infrastructure.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=/var/folders/51/ct25jx3x6xn53_3s8r5449qr0000gn/T/tmpHOliq8.tmp /verbosity:quiet /nologo /Users/ademclk/RiderProjects/ceyehat-api/Ceyehat.Infrastructure/Ceyehat.Infrastructure.csproj
Writing '/Users/ademclk/RiderProjects/ceyehat-api/ceyehat/obj/ceyehat.csproj.EntityFrameworkCore.targets'...
dotnet msbuild /target:GetEFProjectMetadata /property:EFProjectMetadataFile=/var/folders/51/ct25jx3x6xn53_3s8r5449qr0000gn/T/tmpfJx0jN.tmp /verbosity:quiet /nologo /Users/ademclk/RiderProjects/ceyehat-api/ceyehat/ceyehat.csproj
Build started...
dotnet build /Users/ademclk/RiderProjects/ceyehat-api/ceyehat/ceyehat.csproj /verbosity:quiet /nologo

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.36
Build succeeded.
dotnet exec --depsfile /Users/ademclk/RiderProjects/ceyehat-api/ceyehat/bin/Debug/net7.0/ceyehat.deps.json --additionalprobingpath /Users/ademclk/.nuget/packages --runtimeconfig /Users/ademclk/RiderProjects/ceyehat-api/ceyehat/bin/Debug/net7.0/ceyehat.runtimeconfig.json /Users/ademclk/.dotnet/tools/.store/dotnet-ef/7.0.3/dotnet-ef/7.0.3/tools/net6.0/any/tools/netcoreapp2.0/any/ef.dll migrations add PassengerMapping --assembly /Users/ademclk/RiderProjects/ceyehat-api/ceyehat/bin/Debug/net7.0/Ceyehat.Infrastructure.dll --project /Users/ademclk/RiderProjects/ceyehat-api/Ceyehat.Infrastructure/Ceyehat.Infrastructure.csproj --startup-assembly /Users/ademclk/RiderProjects/ceyehat-api/ceyehat/bin/Debug/net7.0/ceyehat.dll --startup-project /Users/ademclk/RiderProjects/ceyehat-api/ceyehat/ceyehat.csproj --project-dir /Users/ademclk/RiderProjects/ceyehat-api/Ceyehat.Infrastructure/ --root-namespace Ceyehat.Infrastructure --language C# --framework net7.0 --nullable --working-dir /Users/ademclk/RiderProjects/ceyehat-api --verbose
Using assembly 'Ceyehat.Infrastructure'.
Using startup assembly 'ceyehat'.
Using application base '/Users/ademclk/RiderProjects/ceyehat-api/ceyehat/bin/Debug/net7.0'.
Using working directory '/Users/ademclk/RiderProjects/ceyehat-api/ceyehat'.
Using root namespace 'Ceyehat.Infrastructure'.
Using project directory '/Users/ademclk/RiderProjects/ceyehat-api/Ceyehat.Infrastructure/'.
Remaining arguments: .
Finding DbContext classes...
Finding IDesignTimeDbContextFactory implementations...
Finding application service provider in assembly 'ceyehat'...
Finding Microsoft.Extensions.Hosting service provider...
Using environment 'Development'.
Using application service provider from Microsoft.Extensions.Hosting.
Found DbContext 'CeyehatDbContext'.
Finding DbContext classes in the project...
Using context 'CeyehatDbContext'.

```

### Include provider and version information

EF Core version:
Database provider: Microsoft.EntityFrameworkCore.PostgreSQL
Target framework: .NET 7.0
Operating system: macOS
IDE: (Rider 2022.3.2)

**Comment by ajcvickers** [comment]
@ademclk Can you either attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing, or, failing that, provide steps to reproduce the error using your full codebase?

**Comment by ademclk** [comment]
> @ademclk Can you either attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing, or, failing that, provide steps to reproduce the error using your full codebase?

Actually looking into whole project might be better because they're entities related each other. In infrastructure layer I have Persistence folder with contains configurations and repositories. Other configurations didn't cause this issue while I was using them for example. ```dotnet ef migrations add AirlineMapping -p Ceyehat.Infrastructure -s ceyehat```

Now if you clone the repository hard reset to this commit 51f251ae975dc6da416f5d34e24736e5db4817cb
and run dotnet ef migrations it'll work fine. Problem is mapping the passenger entity owns many method gives an error I did the same thing what I did with other entities.

I appreciate your help.
